ViZDoomPPO/load_model_generate_dataset.py

```diff
@@ -422,73 +421,149 @@ def make_gif(agent, file_path, num_episodes=1):
-            "images": Sequence(Image()),
-        }
-    )
-
-    # Create the dataset
-    dataset = Dataset.from_list(entries, features=features)
+            buffer['health'].append(int(health))
+            buffer['actions'].append(int(action[0]))
+            buffer['images'].append(screen)
 
-    # Save to Parquet
-    dataset.to_parquet(parquet_path)
+            if len(buffer['health']) == buffer_size:
+                processed_buffer = process_buffer(buffer, episode, step_id)
+                pkl_path = os.path.join(episode_dir, f"buffer_{step_id}.pkl")
+                with open(pkl_path, 'wb') as f:
+                    pickle.dump(processed_buffer, f)
+                step_id += 1
 
-    print(f"Data saved to {parquet_path}")
     env.close()
 
-
-MODEL_PATH = "/Users/pbowmandavis/Downloads/gameNgen-repro/stable_baselines_playground/rl-doom/standalone_examples/logs/models/dm_longrun_test/final_model.zip"
+def concatenate_pkls_to_parquet(input_dir, output_parquet, batch_size=1000):
+    schema = pa.schema([
+        ('episode_id', pa.int32()),
+        ('step_id', pa.int32()),
+        ('health', pa.list_(pa.int32())),
+        ('actions', pa.list_(pa.int32())),
+        ('images', pa.list_(pa.string()))
+    ])
+
+    writer = None
+    batch = []
+
+    for episode_dir in tqdm(os.listdir(input_dir), desc="Processing episodes"):
+        episode_path = os.path.join(input_dir, episode_dir)
+        if os.path.isdir(episode_path):
+            for pkl_file in os.listdir(episode_path):
+                if pkl_file.endswith('.pkl'):
+                    with open(os.path.join(episode_path, pkl_file), 'rb') as f:
+                        data = pickle.load(f)
+                        batch.append(data)
+
+                    if len(batch) >= batch_size:
+                        df = pd.DataFrame(batch)
+                        table = pa.Table.from_pandas(df, schema=schema)
+
+                        if writer is None:
+                            writer = pq.ParquetWriter(output_parquet, schema)
+
+                        writer.write_table(table)
+                        batch = []
+
+    # Write any remaining data
+    if batch:
+        df = pd.DataFrame(batch)
+        table = pa.Table.from_pandas(df, schema=schema)
+        if writer is None:
+            writer = pq.ParquetWriter(output_parquet, schema)
+        writer.write_table(table)
+
+    if writer:
+        writer.close()
+
+    print(f"Data saved to {output_parquet}")
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="Generate GIF or Parquet file from pretrained PPO agent")
+    parser.add_argument("--output", choices=["gif", "parquet"], required=True, help="Output format")
+    parser.add_argument("--episodes", type=int, default=1, help="Number of episodes to run")
+    parser.add_argument("--upload", action="store_true", help="Upload the output to Hugging Face Hub")
+    parser.add_argument("--hf_token", help="Hugging Face API token")
```

Comment:
You can usually use the HF_TOKEN environement variable for this, so I'd make this an optional argument